libraries/AP_Baro/AP_Baro.cpp

```diff
@@ -889,6 +916,22 @@ void AP_Baro::update(void)
         }
     }
 
+    const uint32_t now_ms = AP_HAL::millis();
+    if (now_ms - _field_elevation_last_ms >= 1000 && fabs(_field_elevation_active-_field_elevation) > 1.0) {
+      if (!AP::arming().is_armed()) {
```

Comment:
Why not when armed?

Reply:
This is the 1.0 version of this algorithm to get us flying.  It currently automatically recalibrates the pressure to set the measured barometric altitude such that your altitude is zero.  If you were to do this in flight, it would immediately set your pressure calibration to zero altitude.  The primary goal of this was to fix the barometric altitude calculation and not necessarily improve the altitude user experience for doing things like changing your altitude reference in flight from METAR etc.  That is going to be the 2.0 version of this approach.  However, getting this version out quickly will support our current flight ops where we have to be in compliance with General Aviation atmospheric model.